backend/management/commands/config.py

```diff
@@ -0,0 +1,231 @@
+from django.core.management.base import BaseCommand
+from backend.models import Backend
+from backend.models import Example
+import yaml
+
+
+class Command(BaseCommand):
+    help = (
+        "Usage: python manage.py config <backend> <field> <value>\n"
+        "\n"
+        "Update the specified field of the specified backend to the specified value"
+    )
+
+    # Constructor.
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.log_messages = []
+
+    # Custom log function.
+    def log(self, msg=""):
+        print(msg)
+        self.log_messages.append(msg)
+
+    # Command line arguments.
+    def add_arguments(self, parser):
+        parser.add_argument("backend_slug", help="Slug of the backend")
+        parser.add_argument(
+            "config_file",
+            nargs="?",
+            help="If provided, set the backend config " "according to this YAML file",
+        )
+        parser.add_argument(
+            "--hide-all-other-backends",
+            action="store_true",
+            default=False,
+            help="Hide all other backends (set sort key to 0)",
+        )
+
+    # Return a YAML string for the given dictionary. Format values with
+    # newlines using the "|" style.
+    def dict_to_yaml(self, dictionary):
+        # Custom representer for yaml, which uses the "|" style only for
+        # multiline strings.
+        #
+        # NOTE: We eeplace all `\r\n` with `\n` because otherwise the `|` style
+        # does not work as expected.
+        class MultiLineDumper(yaml.Dumper):
+            def represent_scalar(self, tag, value, style=None):
+                value = value.replace("\r\n", "\n")
+                if isinstance(value, str) and "\n" in value:
+                    style = "|"
+                return super().represent_scalar(tag, value, style)
+
+        # Dump as yaml.
+        return yaml.dump(
+            dictionary,
+            sort_keys=False,
+            Dumper=MultiLineDumper,
+        )
+
+    # Get backend config and examples for the given `backend_slug`. Return the
+    # result as a nicely formatted YAML string.
+    def get_backend_config(self, backend_slug):
+        backend = Backend.objects.filter(slug=backend_slug).get()
+        backend_id = backend.id
+
+        # Get the `Backend` config for this backend.
+        backend_config = {}
+        for field in Backend._meta.get_fields():
+            value = getattr(backend, field.name, None)
+            if value is not None:
+                backend_config[field.name] = str(value)
```

Comment:
suggestion: Is it possible to use the `django.forms.models.model_to_dict` ([source](https://docs.djangoproject.com/en/5.0/_modules/django/forms/models/)) function here? It converts a model instance to a dict. 
question: django has functionality for [serializing model objects](https://docs.djangoproject.com/en/5.1/topics/serialization/#yaml) into some formats. Are you aware of those (though it seems that they go directly to string, which wouldn't help us)?

Reply:
I am aware, but what we want here is serialization to and from YAML (we want a local config file, which the user can edit)